Trappist/src/Promact.Trappist.Repository/Test/TestsRepository.cs

```diff
@@ -333,7 +334,20 @@ await _dbContext.TestQuestion
 
         public async Task<TestAC> GetTestByLinkAsync(string link)
         {
+
             var test = await _dbContext.Test.SingleAsync(x => x.Link.Equals(link));
+
+            var testAttendeeDetails = await _dbContext.TestAttendees.Where(x => x.TestId == test.Id).ToListAsync();
+            foreach (var attendee in testAttendeeDetails)
+            {
+                var testLogs = new TestLogs();
+                testLogs = await _dbContext.TestLogs.Where(x => x.TestAttendeeId == attendee.Id).FirstOrDefaultAsync();
+                if(testLogs.StartTest == default(DateTime))
+                testLogs.StartTest = DateTime.UtcNow;
+                _dbContext.TestLogs.Update(testLogs);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') This line will not be executed conditionally; only the first line of this 2-line block will be. The rest will execute unconditionally. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS2681)